Cookbook/Proposals/SwiftGen.md

```diff
@@ -0,0 +1,83 @@
+## Adopt SwiftGen for localizable strings
+
+* Author: David Rodrigues
+* Review Manager: Danilo Aliberti
+
+### Introduction
+
+Localisable strings by themselves are not a hard thing to use but for apps with a significant number of strings it's entirely impossible to remember or discover easily the respective key to use.
+
+```swift
+title = NSLocalizedString("hello_world_title", comment: "")
+```
+
+Because of this, and considering the size of our codebase, we use concrete types that map our localisable strings which enables the usage of autocomplete.
+
+```swift
+public struct LocalizationUI {
+    public struct Common {
+        public static var ok: String { return localizedName(key: "general_ok") }
+        ...
+    }
+}
+```
+
+### Problems
+
+The current approach works, as we would expect, but there's a couple of caveats in it.
+
+##### 1. Maintenance
+
+This types at the moment are created and managed by us manually which means any new addition/edition/removal needs to be done by us. This may not look very significant but considering the number of strings, more than 3k, it is a lot to be maintained/extended.
```

Comment:
```suggestion
These types at the moment are created and managed by us manually which means any new addition/change/removal needs to be done by us. This may not look very significant but considering the number of strings, more than 3k, it is a lot to be maintained/extended.
```